workshop.wwwapi/Endpoints/SurgeryEndpoint.cs

```diff
@@ -1,24 +1,52 @@
 ﻿using Microsoft.AspNetCore.Mvc;
 using workshop.wwwapi.Repository;
+using workshop.wwwapi.Models;
 
 namespace workshop.wwwapi.Endpoints
 {
     public static class SurgeryEndpoint
     {
         //TODO:  add additional endpoints in here according to the requirements in the README.md 
+        public class PatientResponseDTO
+        {
+            public int ID {  get; set; }
+            public string name { get; set; }
+
+            public PatientResponseDTO(Patient patient)
+            {
+                name = patient.FullName;
+                ID = patient.Id;
+            }
+        }
         public static void ConfigurePatientEndpoint(this WebApplication app)
         {
             var surgeryGroup = app.MapGroup("surgery");
 
             surgeryGroup.MapGet("/patients", GetPatients);
+            surgeryGroup.MapGet("/getPatientByID", GetPatientByID);
             surgeryGroup.MapGet("/doctors", GetDoctors);
             surgeryGroup.MapGet("/appointmentsbydoctor/{id}", GetAppointmentsByDoctor);
         }
         [ProducesResponseType(StatusCodes.Status200OK)]
         public static async Task<IResult> GetPatients(IRepository repository)
-        { 
-            return TypedResults.Ok(await repository.GetPatients());
+        {
+            var patient = await repository.GetPatients();
+            var results = new List<PatientResponseDTO>();
+            foreach (var pat in patient)
+            {
+                results.Add(new PatientResponseDTO(pat));
+            }
+            return TypedResults.Ok(results);
         }
+
+        [ProducesResponseType(StatusCodes.Status200OK)]
+        public static async Task<IResult> GetPatientByID(IRepository repository, int id)
+        {
+            var patient = await repository.GetPatientByID(id);
```

Comment:
you should return NotFound if no patient with id exists

Reply:
resolved